py/foundation/in_memory.py

```diff
@@ -217,6 +221,57 @@ def __len__(self) -> int:
   def from_dict(d: Dict[str, Any]) -> 'InMemoryText':
     ...
 
+@attr.s(auto_attribs=True)
+class InMemorySpatialText(Text):
+  _id: str
+  _lines: List[InMemoryText]
+
+  @property
+  def type(self) -> str:
+    return "SpatialText"
+  
+  @property
+  def id(self) -> str:
+    return self._id
+
+  def _get_in_memory_text(self) -> 'InMemoryText':
+    # Gets a one dimensional InMemoryText
+    full_list: List[Union[Word, Whitespace]] = list(self._lines[0].get_children_with_spaces())
+    new_line = InMemoryWhitespace(text='\n')
+    for i in range(1, len(self._lines)):
+      full_list.append(new_line)
```

Comment:
Is there a chance this creates problems later where all objects reference the same `\n` object? Would it be better to create a new one for each instead of a single shared newline reference?

Reply:
Good point. I'll patch that up and make a new Whitespace instance for each newline.